vars/artifactSetVersion.groovy

```diff
@@ -9,6 +9,8 @@ import com.sap.piper.versioning.ArtifactVersioning
 import groovy.transform.Field
 import groovy.text.SimpleTemplateEngine
 
+enum GitPushMode {NONE, HTTPS, SSH}
```

Comment:
Why `NONE`? We only have an `else`, so that can be anything or unset ...

Reply:
With `NONE`we can make the commit with out directly pushing it. Seems reasonable to me to first make the version change, build afterwards and push the commit when the build succeeded. Techically it is also possible to only make the version change without commit. But maybe some tools incorporate the git commit hash somewhere in the binaries (e.g. for jars: `META-INF/MANIFEST.mf`. In this case we get the commit without the version change, but what was actually built was the commit with the version change.